_posts/2019-08-27-bazel-0.29.0.md

```diff
@@ -0,0 +1,56 @@
+*   The [platform](https://docs.bazel.build/versions/master/be/platform.html#platform) attribute `remote_execution_properties` has been deprecated in favor of the newly introduced `exec_properties` attribute.
+
+
+## Java
+
+*   We fixed remote execution on Windows for the [Java rules](https://docs.bazel.build/versions/master/be/java.html). See [#8614](https://github.com/bazelbuild/bazel/issues/8614).
+*   You can now use a new [`java_toolchain`](https://docs.bazel.build/versions/master/be/java.html#java_toolchain) target `@remote_java_tools_{linux/windows/darwin}//:prebuilt_toolchain` in your builds. The new toolchain uses only pre-built tools, including singlejar and ijar, even on remote execution. The prebuilt toolchain  should be used only when host and execution platform are the same, otherwise the binaries will not work on the execution platform.
+*   [`java_common.compile`](https://docs.bazel.build/versions/master/skylark/lib/java_common.html#compile) supports specifying `annotation_processor_additional_inputs` and `annotation_processor_additional_outputs`. The API for the Java compilation action enables supporting annotation processors that consume or produce artifacts.
+
+
+## Android
+
+*   The Android NDK is now integrated with toolchains. To use them, pass the `--extra_toolchains=@androidndk//:all` flag or register them in your WORKSPACE with `register_toolchains("@androidndk//:all")`. Learn more [here](https://docs.bazel.build/versions/master/android-ndk.html#integration-with-platforms-and-toolchains).
+*   There is now documentation on optimizing Android app build performance. Read it [here](https://docs.bazel.build/versions/0.29.0/android-build-performance.html).
+
+
+## Python
+
+*   Added `--incompatible_load_python_rules_from_bzl`, which will be flipped in a future Bazel version. To migrate, load Python rules from `@rules_python`. See [#9006](https://github.com/bazelbuild/bazel/issues/9006).
+
+
+## Other
+
+*   Add new incompatible flag `--incompatible_load_proto_rules_from_bzl` to forbid loading the native proto rules. For migration instructions, see tracking issue [#8922](https://github.com/bazelbuild/bazel/issues/8922)
+*   `rule_test`: fix Bazel 0.27 regression ("tags" attribute was ignored, [#8723](https://github.com/bazelbuild/bazel/issues/8723))
+*   The `objc_proto_library` rule has been [deleted](https://github.com/bazelbuild/bazel/issues/7348) from Bazel.
+*   `@build_tools/tools/build_defs/pkg:{pkg_deb, pkg_rpm, pkg_tar}` are documented as deprecated. These features are now available independently of Bazel releases on the [`rules_pkg` repository](https://github.com/bazelbuild/rules_pkg/tree/master/pkg). The original rules will continue to exist until Bazel 2.0.
+*   Added a new flag `--//tools/build_defs/pkg:incompatible_no_build_defs_pkg` to forbid loading rules from `//tools/build_defs/pkg:{pkg_deb, pkg_rpm, pkg_tar}`.
```

Comment:
@aiuto : Please verify the flag correct. Did you mean `--//tools/build_defs/pkg:incompatible_no_build_defs_pkg` or `--incompatible_no_build_defs_pkg`?

Reply:
Also, the `@build_tools/tools/build_defs/pkg:{pkg_deb, pkg_rpm, pkg_tar}` are not valid, I suggest writing it as:

> `@build_tools/tools/build_defs/pkg:pkg_deb`, `:pkg_rpm`, and `:pkg_tar`

Reply:
Also, does this flag have any effect outside of the Bazel source tree? (Does it require that `//tools/build_defs/pkg` be a valid package in the source tree?)